### PR TITLE
Add html output for graph build-order and graph build-order-merge

### LINKED ISSUE #16356: [feature] html graph from existing build-order.json
### What is your suggestion?

The `conan graph info` command has a fantastic html output for graphs that can be used to look through a chosen graph.
The `conan graph build-order` command can be used to produce many such chunks of graphable data,
and the `conan graph build-order-merge` command does a great job combining several into a single json file,..

...But there does not seem to be any way to take a merged build-order json file, and feed it into the same nice html format for review. If there is one, I haven't found it yet. :)

I would love it if, either:

1. The `build-order` and/or `build-order-merge` commands, also supported html output (I would just output my json and then my html with the same command),
_or_
2. If the `conan graph info` command, could support taking in an input json file, and graphing that.

I don't have any preference, both versions would satisfy my needs just fine, and I'm happy to entertain other alternatives that I missed when evaluating this feature request.

If there was a chosen method from above, or another, I would not be totally against attempting to contribute such a method, though my familiarity with the conan codebase itself is low at the moment.

Thanks for considering!

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

### PR DESCRIPTION
**Pull request by czoido**
Changelog: Feature: Add html output for graph build-order and graph build-order-merge
Docs: https://github.com/conan-io/docs/pull/3805

Closes: https://github.com/conan-io/conan/issues/16356

image

### CODE REVIEW

**Review by memsharded — APPROVED** [review]
I think it is a pity that we couldnt do this with the other js library, and use the same node formatters. If possible I'd try to use tool-requires and test-requires with a different shape, but not a big issue. 
Also the alignment of the layers, but I guess that you have already tried and it is far from easy, so not an issue either.

Good to go!

**Comment by czoido** [comment]
> I think it is a pity that we couldnt do this with the other js library, and use the same node formatters. If possible I'd try to use tool-requires and test-requires with a different shape, but not a big issue. Also the alignment of the layers, but I guess that you have already tried and it is far from easy, so not an issue either.
> 
> Good to go!

Yes, it looks like _vis.js_ is not the right library for this kind of diagrams... :face_with_diagonal_mouth:
I have updated the output to show the requires from the build context a bit different.

File: conan/cli/formatters/graph/build_order_html.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+                width: 20px;
+                height: 20px;
+                margin-right: 10px;
+            }
+        </style>
+
+        <div class="sidebar">
+            <div class="legend">
+                <div class="legend-item">
+                    <div class="legend-color" style="background-color: #FFCC80;"></div>
```
Comment (AbrilRBS) [inline conan/cli/formatters/graph/build_order_html.py]: Can this colour have a bit more contrast with "Cache" ones? It's a bit hard for my fellow colourblind people to distiguish them apart

File: conan/cli/formatters/graph/build_order_html.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+                        if (Array.isArray(lib.packages)) {
+                            lib.packages.forEach(pkgArray => {
+                                pkgArray.forEach(pkg => {
+                                    if (pkg.binary === "Build") {
+                                        isSomeBuild = true;
+                                    } else {
+                                        isAllBuild = false;
+                                    }
+                                    if (pkg.context === "build") {
+                                        shape = 'cut-rectangle';
```
Comment (AbrilRBS) [inline conan/cli/formatters/graph/build_order_html.py]: Should we follow the same legend as the `conan graph info` output?

**Review by AbrilRBS — COMMENTED** [review]
Few minor comments, otherwise looks good

File: conan/cli/formatters/graph/build_order_html.py (hunk 2 above)
Comment (czoido) [reply conan/cli/formatters/graph/build_order_html.py]: I updated the colors and node type to be more coherent with the graph info, can you check?

**Review by AbrilRBS — APPROVED** [review]

**Pull request merged**